src/mechanics/constraintsPde/ConstraintCompanion.cpp

```diff
@@ -0,0 +1,106 @@
+#include "mechanics/constraints/ConstraintCompanion.h"
+#include "mechanics/groups/Group.h"
+
+namespace NuTo
+{
+namespace Constraint
+{
+
+std::vector<Equation> Component(const NodeBase& node, std::vector<eDirection> directions, double value)
+{
+    return Component(node, directions, RhsConstant(value));
+}
+
+
+std::vector<Equation> Component(const NodeBase& node, std::vector<eDirection> directions, RhsFunction rhs)
+{
+    std::vector<Equation> eqs;
+    for (auto direction : directions)
+    {
+        int component = ToComponentIndex(direction);
+        if (node.GetNumDofs() < component)
+            // TODO This check is not meaningful at the moment, since this method returns the
+            // total number of dofs, say 4 (3 disp, 1 temp). This will not find the error
+            // if you try to constrain the Z component of the temperature. Which would be wrong.
+            throw Exception(__PRETTY_FUNCTION__, "Dimension mismatch");
+
+        eqs.push_back(Equation({Term(node, component, 1)}, rhs));
+    }
+    return eqs;
+}
+
+
+std::vector<Equation> Component(const Group<NodeBase>& nodes, std::vector<eDirection> directions, double value)
+{
```

Comment:
I think you will use the new Group here, yes?

Reply:
The ConstraintCompanion is a point on the bucket list above. Will fix that soon.